### PR TITLE
clarify pull option

### PR DESCRIPTION
**Pull request by oceanBT**

### CODE REVIEW

**Comment by crazy-max** [comment]
Don't think that's necessary. Also it does not only concern `FROM` instructions but also:
* `# syntax=<image>`
* `COPY --from=<image>`
* `RUN --mount=type=bind,from=<image>`

**Comment by oceanBT** [comment]
It took me 30 minutes to figure out exactly what is intended by "the image". Others seem to have had a similar experience (#55). 

> Also it does not only concern `FROM` instructions but also:
> 
> * `# syntax=<image>`
> * `COPY --from=<image>`
> * `RUN --mount=type=bind,from=<image>`

Thanks for this advice! I have not known that yet.

How about: 

> pull all images that are referenced in the Dockerfile (e. g. `FROM`, `COPY`) during build

**Comment by crazy-max** [comment]
@oceanBT 

`Always attempt to pull a newer version of all images that are referenced in the Dockerfile` sgtm.

Might need [changes upstream](https://github.com/docker/buildx/blob/c643c2ca954c19f2a73601745e14d69967bef583/commands/build.go#L367) to keep it consistent. PTAL @tonistiigi @thaJeztah

**Comment by thaJeztah** [comment]
LOL, I started to write a comment, but opened a can of worms while writing it. Anyway; here's what I wrote 😅 


> Always attempt to pull a newer version of all images that are referenced in the Dockerfile

SGTM; perhaps "newer versions"

```
Always attempt to pull newer versions of images that are referenced in the Dockerfile.
```

Unless we want to make it more generic (not mention Dockerfile?)

```
Always attempt to pull newer versions of the images images used by the build.
```


Also looking a bit at "attempt to" (it's a bit ambiguous, as it may mean either "try to pull, but don't error if it fails", or "if a newer image is available, pull it")

```
Always pull newer versions of the images images used by the build.
```

Which brings it to "what is _newer_ ? " (because it will pull the same tag, not "update to a newer version") 🤔

File: README.md [hunk 1]
```diff
@@ -210,7 +210,7 @@ Following inputs can be used as `step.with` keys
 | `target`            | String   | Sets the target stage to build |
 
 ### outputs
-
+
```
Comment (crazy-max) [inline README.md:213]: Revert this change

**Review by crazy-max — CHANGES_REQUESTED** [review]
Check DCO (sign-off is missing)

File: README.md [hunk 2]
```diff
@@ -201,7 +201,7 @@ Following inputs can be used as `step.with` keys
 | `no-cache`          | Bool     | Do not use cache when building the image (default `false`) |
 | `outputs`           | List     | List of [output destinations](https://github.com/docker/buildx/blob/master/docs/reference/buildx_build.md#output) (format: `type=local,dest=path`) |
 | `platforms`         | List/CSV | List of [target platforms](https://github.com/docker/buildx/blob/master/docs/reference/buildx_build.md#platform) for build |
-| `pull`              | Bool     | Always attempt to pull a newer version of the image (default `false`) |
```
Comment (crazy-max) [inline README.md:204]: Please add back

```
(default `false`)
```

**Review by crazy-max — CHANGES_REQUESTED** [review]
And squash your commits, thanks.

**Pull request closed without merging**